pkg/preparation/uploads/uploads.go

```diff
@@ -152,66 +143,95 @@ func (e executor) execute(ctx context.Context) (cid.Cid, error) {
 	return e.upload.RootCID(), err
 }
 
-func (e *executor) runScanWorker(ctx context.Context, dagWork chan<- struct{}) error {
-	log.Debugf("Running new scan for upload %s in state %s", e.upload.ID(), e.upload.State())
+// runStartWorker runs the worker that creates the scan and signals that it's
+// available to run.
+func (e *executor) runStartWorker(ctx context.Context, uploadAvailable <-chan struct{}, scansAvailable chan<- struct{}) error {
+	return Worker(
+		ctx,
+		uploadAvailable,
 
-	// Unlike later stages, this one doesn't need to watch a work channel with
-	// [Worker], because it never has to wait for work.
+		// doWork
+		func() error {
+			if e.upload.State() == model.UploadStatePending {
+				if err := e.upload.Start(); err != nil {
+					return fmt.Errorf("starting upload: %w", err)
+				}
+				if err := e.api.Repo.UpdateUpload(ctx, e.upload); err != nil {
+					return fmt.Errorf("updating upload: %w", err)
+				}
+				if _, err := e.api.Repo.CreateScan(ctx, e.upload.ID()); err != nil {
```

Comment:
vaguely expect to update the upload AFTER the scan is created? Not sure why though.

Reply:
I guess ideal would be to transactionalize the whole thing but... meh